src/main.rs

```diff
@@ -20,20 +21,28 @@ fn main() -> anyhow::Result<()> {
         _ => "trace",
     })).init();
 
-    let remote_name = args.remote + "/";
-    let remote_name = remote_name.as_str();
     let repo_path = args.repo.as_deref().unwrap_or(".");
     let repo = open_repo(repo_path)?;
 
+    // Obtain remote name from args or use default
+    let remote_name = if let Some(name) = args.remote { name } else {
+        let remotes = repo.remotes()?;
+        if remotes.len() != 1 {
+            return Err(anyhow!(AppError::RemoteNotSpecified));
+        }
+        remotes.get(0).unwrap().to_string()
+    };
+    let remote_name_prefix = remote_name.to_string() + "/";
+
     let branch_names = repo.branches(None)?
         .try_filter_map(|(branch, _)| {
             // get name when not Error
             branch.name().map(|name| {
                 // get name when not None
                 name.map(|name| {
                     if branch.get().is_remote() {
-                        name.strip_prefix(remote_name)
-                            .map(|name| name.to_string())   // get own string when stripped
+                        name.strip_prefix(&remote_name_prefix)
```

Comment:
&String can be automated transformed to be a &str?

Reply:
yes. it called [defref corcions](https://doc.rust-lang.org/book/ch15-02-deref.html?#implicit-deref-coercions-with-functions-and-methods)